contracts/mixnet/src/queries.rs

```diff
@@ -0,0 +1,187 @@
+// settings for pagination
+use crate::state::MixNodeBond;
+use crate::state::PREFIX_MIXNODES;
+use cosmwasm_std::Deps;
+use cosmwasm_std::HumanAddr;
+use cosmwasm_std::Order;
+use cosmwasm_std::StdResult;
+use cosmwasm_storage::bucket_read;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+const MAX_LIMIT: u32 = 30;
+const DEFAULT_LIMIT: u32 = 10;
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize, JsonSchema)]
+pub struct PagedResponse {
+    pub nodes: Vec<MixNodeBond>,
+    pub per_page: usize,
+    pub start_next_after: Option<HumanAddr>,
+}
+
+pub fn query_mixnodes_paged(
+    deps: Deps,
+    start_after: Option<HumanAddr>,
+    limit: Option<u32>,
+) -> StdResult<PagedResponse> {
+    let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+    let start = calculate_start_value(start_after);
+
+    let bucket = bucket_read::<MixNodeBond>(deps.storage, PREFIX_MIXNODES);
+    let res = bucket
+        .range(start.as_deref(), None, Order::Ascending)
+        .take(limit);
+    let node_tuples = res.collect::<StdResult<Vec<(Vec<u8>, MixNodeBond)>>>()?;
+    let nodes = node_tuples.into_iter().map(|item| item.1).collect();
```

Comment:
I don't really understand the logic here. Firsts you iterate through `bucket` and collect results into `node_tuples` which you immediately turn into another iterator. Why not skip assignment to `node_tuples` and use `.map` directly?